tests/trial_tests/test_trial.py

```diff
@@ -259,64 +254,72 @@ def test_suggest_discrete_uniform(storage_init_func: Callable[[], storages.BaseS
-
-    with patch.object(
-        distributions, "_get_single_value", wraps=distributions._get_single_value
-    ) as mock_object:
-        assert trial.suggest_uniform("a", 1.0, 1.0) == 1.0  # Suggesting a param.
-        assert mock_object.call_count == 1
-        assert trial.suggest_uniform("a", 1.0, 1.0) == 1.0  # Suggesting the same param.
-        assert mock_object.call_count == 1
+    with StorageSupplier(storage_mode) as storage:
+        study = create_study(storage=storage, sampler=samplers.TPESampler(n_startup_trials=0))
 
-        assert trial.suggest_loguniform("b", 1.0, 1.0) == 1.0  # Suggesting a param.
-        assert mock_object.call_count == 2
-        assert trial.suggest_loguniform("b", 1.0, 1.0) == 1.0  # Suggesting the same param.
-        assert mock_object.call_count == 2
+        trial = Trial(study, study._storage.create_new_trial(study._study_id))
 
-        assert trial.suggest_discrete_uniform("c", 1.0, 1.0, 1.0) == 1.0  # Suggesting a param.
-        assert mock_object.call_count == 3
-        assert (
-            trial.suggest_discrete_uniform("c", 1.0, 1.0, 1.0) == 1.0
-        )  # Suggesting the same param.
-        assert mock_object.call_count == 3
-
-        assert trial.suggest_int("d", 1, 1) == 1  # Suggesting a param.
-        assert mock_object.call_count == 4
-        assert trial.suggest_int("d", 1, 1) == 1  # Suggesting the same param.
-        assert mock_object.call_count == 4
-
-        assert trial.suggest_float("e", 1.0, 1.0) == 1.0  # Suggesting a param.
-        assert mock_object.call_count == 5
-        assert trial.suggest_float("e", 1.0, 1.0) == 1.0  # Suggesting the same param.
-        assert mock_object.call_count == 5
-
-        assert trial.suggest_float("f", 0.5, 0.5, log=True) == 0.5  # Suggesting a param.
-        assert mock_object.call_count == 6
-        assert trial.suggest_float("f", 0.5, 0.5, log=True) == 0.5  # Suggesting the same param.
-        assert mock_object.call_count == 6
-
-        assert trial.suggest_float("g", 0.5, 0.5, log=False) == 0.5  # Suggesting a param.
-        assert mock_object.call_count == 7
-        assert trial.suggest_float("g", 0.5, 0.5, log=False) == 0.5  # Suggesting the same param.
-        assert mock_object.call_count == 7
-
-        assert trial.suggest_float("h", 0.5, 0.5, step=1.0) == 0.5  # Suggesting a param.
-        assert mock_object.call_count == 8
-        assert trial.suggest_float("h", 0.5, 0.5, step=1.0) == 0.5  # Suggesting the same param.
-        assert mock_object.call_count == 8
-
-        assert trial.suggest_int("i", 1, 1, log=True) == 1  # Suggesting a param.
-        assert mock_object.call_count == 9
-        assert trial.suggest_int("i", 1, 1, log=True) == 1  # Suggesting the same param.
-        assert mock_object.call_count == 9
-
-
-@parametrize_storage
+        with patch.object(
+            distributions, "_get_single_value", wraps=distributions._get_single_value
+        ) as mock_object:
```

Comment:
How about moving this `with patch.object ...` to the above and combining `with StorageSupplier...` like as follows? This will reduce the number of changes (indents).
```python
with patch.object(
    distributions, "_get_single_value", wraps=distributions._get_single_value
) as mock_object, StorageSupplier(storage_mode) as storage:
```

Reply:
Thansk!